examples/src/write_model.rs

```diff
@@ -27,8 +27,26 @@ enum Format {
 fn main() {
     let args = Args::parse();
     let model = args.example.shape();
-    match args.format {
-        Format::Step => model.write_step(args.output.with_extension("step")).unwrap(),
-        Format::Stl => model.write_stl(args.output.with_extension("stl")).unwrap(),
+
+    let format = args.format.unwrap_or_else(|| {
+        let extension = args.output.extension().unwrap_or_else(|| {
+            panic!("Cannot guess format because the output file name has no extension.")
+        });
+
+        determine_format(extension)
+            .unwrap_or_else(|| panic!("Cannot guess format from extension {:?}.", extension))
+    });
+
+    match format {
+        Format::Step => model.write_step(args.output).unwrap(),
+        Format::Stl => model.write_stl(args.output).unwrap(),
+    }
+}
+
+fn determine_format(extension: &OsStr) -> Option<Format> {
+    match extension.as_bytes() {
+        b"step" | b"stp" => Some(Format::Step),
+        b"stl" => Some(Format::Stl),
```

Comment:
I'm gonna be that annoying guy and say we should probably do a case-insensitive comparison here.

Reply:
Done! :)